ports/stm/peripherals/stm32f4/stm32f405xx/periph.c

```diff
@@ -192,3 +192,25 @@ const mcu_tim_pin_obj_t mcu_tim_pin_list[56] = {
     // TIM(8,3,2,&pin_PI06),
     // TIM(8,3,3,&pin_PI07),
 };
+
+//SDIO
+SDIO_TypeDef * mcu_sdio_banks[1] = {SDIO};
```

Comment:
Are we future-proofing for a future case where a chip has more than one SDIO bank? I can't grep any instances of SDIO2. The only STM32 boards I'm aware of that have a second SD peripheral are the H7s, which use a different peripheral, the SDMMC driver, which I think would probably deserve its own separate file implementation and pin structure if we get to it. 

There's no real reason to keep the array structure when there's only one instance of the peripheral. I think you could simplify this by simply declaring actual instances of these objects, like the pins. That way you don't need any loops, you just check the passed pins for equivalence directly. It also doesn't imply within the module that a chip can have more than one SDIO when that's not actually possible.

Reply:
It looks like there is some level of intended portability of the same API code to devices which call the hardware block SDMMC:
```
STM32F4xx_HAL_Driver/Inc/stm32f4xx_hal_sd.h:#define SD_TypeDef          SDIO_TypeDef
STM32F7xx_HAL_Driver/Inc/stm32f7xx_hal_sd.h:#define SD_TypeDef          SDMMC_TypeDef
```
.. which goes inside SD_HandleTypeDef, and
```
STM32F4xx_HAL_Driver/Inc/stm32f4xx_hal_sd.h:HAL_StatusTypeDef HAL_SD_Init(SD_HandleTypeDef *hsd);
STM32F7xx_HAL_Driver/Inc/stm32f7xx_hal_sd.h:HAL_StatusTypeDef HAL_SD_Init(SD_HandleTypeDef *hsd);
```

in which case the above should be SD_TypeDef rather than SDIO_TypeDef I guess but otherwise it's pretty close and looks like it'll lay groundwork for devices with SDMMC1 and SDMMC2 peripherals.

Reply:
Ok, we can leave it as is then.

Reply:
Maybe put in a comment that the arrays exist to future proof against the H7